libs/vkd3d/state.c

```diff
@@ -3059,6 +3059,15 @@ static HRESULT d3d12_pipeline_state_init_graphics(struct d3d12_pipeline_state *s
         rt_count = ARRAY_SIZE(graphics->blend_attachments);
     }
 
+    if ((!desc->ps.pShaderBytecode || !desc->ps.BytecodeLength) && rt_count)
+    {
+        /* Avoids validation errors where app might bind bogus RTV format which does not match the PSO.
+         * D3D12 validation does not complain about this when PS is NULL since RTVs are not accessed to begin with.
+         * We can just pretend we have no render targets in this case, which is fine. */
+        WARN("Using render targets with disabled PS. Forcing rt_count = 0.\n");
```

Comment:
Do we really need a `ẀARN` here? I don't think this adds any useful info in a debugging scenario, and at least in D3D11 games this kind of thing is somewhat common.

Reply:
Guess not. Fixed.